.github/workflows/dev_docker_image.yml

```diff
@@ -0,0 +1,39 @@
+name: Docker Image for Development
+on:
+  # should support manual trigger and PRs to the Dockerfile
+  push:
+    paths:
+      - 'docker/**'
+      - '.github/workflows/dev_docker_image.yml'
```

Comment:
I don't see the other jobs depend on `Docker Image for Development` job. So changes in the dev docker creation won't be reflected until the image got pushed in `Docker Image for Development` job.

I feel like the checksum checking method in `cibw_docker_image.yml` for triggering the creation of new docker a better alternative

Reply:
We make changes to this docker very infrequently so I prefer to keep the flows a bit simpler and to not add dependencies between them unless it is needed.
Also this image will be used in other repos as well so it has to be tested manually in any case.

Reply:
Yes I am torn. I understand this docker image won't be changed frequently so it's hard to justify complicating already complicated wheel building process.
At the same time, if any changes made to this docker image, the CI can't tell us whether it works until the change is merged and tag `latest` is updated. This kinda defy the purpose of CI. 
I'll approve it and leave it to Alex for thoughts

Reply:
It would be nice to be able to run the CI with a candidate new image. How about a run parameter to choose the image tag rather than hardcoding `latest`? Would then be easy to try out a CI run with a candidate image

Reply:
good point , I've added a parameter to override the default tag